  <main role="main">    <h1>Class CertificateException</h1>   <ul class="inheritance"> <li><a href="../../lang/object">java.lang.Object</a></li> <li> <ul class="inheritance"> <li><a href="../../lang/throwable">java.lang.Throwable</a></li> <li> <ul class="inheritance"> <li><a href="../../lang/exception">java.lang.Exception</a></li> <li> <ul class="inheritance"> <li><a href="../generalsecurityexception">java.security.GeneralSecurityException</a></li> <li> <ul class="inheritance"> <li>java.security.cert.CertificateException</li> </ul> </li> </ul> </li> </ul> </li> </ul> </li> </ul>   <dl> <dt>All Implemented Interfaces:</dt> <dd><code><a href="../../io/serializable">Serializable</a></code></dd> </dl> <dl> <dt>Direct Known Subclasses:</dt> <dd>
<code><a href="certificateencodingexception">CertificateEncodingException</a></code>, <code><a href="certificateexpiredexception">CertificateExpiredException</a></code>, <code><a href="certificatenotyetvalidexception">CertificateNotYetValidException</a></code>, <code><a href="certificateparsingexception">CertificateParsingException</a></code>, <code><a href="certificaterevokedexception">CertificateRevokedException</a></code>
</dd> </dl>  <pre data-language="java">public class CertificateException
extends GeneralSecurityException</pre> <p>This exception indicates one of a variety of certificate problems.</p> <dl> <dt>Since:</dt> <dd>1.2</dd> <dt>See Also:</dt> <dd>
<a href="certificate"><code>Certificate</code></a>, <a href="https://docs.oracle.com/en/java/javase/11/docs/api/serialized-form.html#java.security.cert.CertificateException">Serialized Form</a>
</dd> </dl>      <section role="region">  <h2 id="constructor.summary">Constructors</h2> <table>  <tr> <th scope="col">Constructor</th> <th scope="col">Description</th> </tr> <tr> <th scope="row"><code><a href="#%3Cinit%3E()">CertificateException</a>()</code></th> <td> <p>Constructs a certificate exception with no detail message.</p> </td> </tr> <tr> <th scope="row"><code><a href="#%3Cinit%3E(java.lang.String)">CertificateException</a>​(<a href="../../lang/string">String</a> msg)</code></th> <td> <p>Constructs a certificate exception with the given detail message.</p> </td> </tr> <tr> <th scope="row"><code><a href="#%3Cinit%3E(java.lang.String,java.lang.Throwable)">CertificateException</a>​(<a href="../../lang/string">String</a> message,
                    <a href="../../lang/throwable">Throwable</a> cause)</code></th> <td> <p>Creates a <code>CertificateException</code> with the specified detail message and cause.</p> </td> </tr> <tr> <th scope="row"><code><a href="#%3Cinit%3E(java.lang.Throwable)">CertificateException</a>​(<a href="../../lang/throwable">Throwable</a> cause)</code></th> <td> <p>Creates a <code>CertificateException</code> with the specified cause and a detail message of <code>(cause==null ? null : cause.toString())</code> (which typically contains the class and detail message of <code>cause</code>).</p> </td> </tr> </table>  </section>  <section role="region">  <h2 id="method.summary">Methods</h2>  <h2 id="methods.inherited.from.class.java.lang.Throwable">Methods declared in class java.lang.<a href="../../lang/throwable">Throwable</a>
</h2> <code><a href="../../lang/throwable#addSuppressed(java.lang.Throwable)">addSuppressed</a>, <a href="../../lang/throwable#fillInStackTrace()">fillInStackTrace</a>, <a href="../../lang/throwable#getCause()">getCause</a>, <a href="../../lang/throwable#getLocalizedMessage()">getLocalizedMessage</a>, <a href="../../lang/throwable#getMessage()">getMessage</a>, <a href="../../lang/throwable#getStackTrace()">getStackTrace</a>, <a href="../../lang/throwable#getSuppressed()">getSuppressed</a>, <a href="../../lang/throwable#initCause(java.lang.Throwable)">initCause</a>, <a href="../../lang/throwable#printStackTrace()">printStackTrace</a>, <a href="../../lang/throwable#printStackTrace(java.io.PrintStream)">printStackTrace</a>, <a href="../../lang/throwable#printStackTrace(java.io.PrintWriter)">printStackTrace</a>, <a href="../../lang/throwable#setStackTrace(java.lang.StackTraceElement%5B%5D)">setStackTrace</a>, <a href="../../lang/throwable#toString()">toString</a></code>  <h2 id="methods.inherited.from.class.java.lang.Object">Methods declared in class java.lang.<a href="../../lang/object">Object</a>
</h2> <code><a href="../../lang/object#clone()">clone</a>, <a href="../../lang/object#equals(java.lang.Object)">equals</a>, <a href="../../lang/object#finalize()">finalize</a>, <a href="../../lang/object#getClass()">getClass</a>, <a href="../../lang/object#hashCode()">hashCode</a>, <a href="../../lang/object#notify()">notify</a>, <a href="../../lang/object#notifyAll()">notifyAll</a>, <a href="../../lang/object#wait()">wait</a>, <a href="../../lang/object#wait(long)">wait</a>, <a href="../../lang/object#wait(long,int)">wait</a></code>  </section>   <div>   <section role="region">  <h2 id="constructor.detail">Constructors</h2>   <h3 id="&lt;init&gt;()">CertificateException</h3> <pre data-language="java">public CertificateException()</pre> <p>Constructs a certificate exception with no detail message. A detail message is a String that describes this particular exception.</p>    <h3 id="&lt;init&gt;(java.lang.String)">CertificateException</h3> <pre data-language="java">public CertificateException(String msg)</pre> <p>Constructs a certificate exception with the given detail message. A detail message is a String that describes this particular exception.</p> <dl> <dt>Parameters:</dt> <dd>
<code>msg</code> - the detail message.</dd> </dl>    <h3 id="&lt;init&gt;(java.lang.String,java.lang.Throwable)">CertificateException</h3> <pre data-language="java">public CertificateException(String message,
                            Throwable cause)</pre> <p>Creates a <code>CertificateException</code> with the specified detail message and cause.</p> <dl> <dt>Parameters:</dt> <dd>
<code>message</code> - the detail message (which is saved for later retrieval by the <a href="../../lang/throwable#getMessage()"><code>Throwable.getMessage()</code></a> method).</dd> <dd>
<code>cause</code> - the cause (which is saved for later retrieval by the <a href="../../lang/throwable#getCause()"><code>Throwable.getCause()</code></a> method). (A <code>null</code> value is permitted, and indicates that the cause is nonexistent or unknown.)</dd> <dt>Since:</dt> <dd>1.5</dd> </dl>    <h3 id="&lt;init&gt;(java.lang.Throwable)">CertificateException</h3> <pre data-language="java">public CertificateException(Throwable cause)</pre> <p>Creates a <code>CertificateException</code> with the specified cause and a detail message of <code>(cause==null ? null : cause.toString())</code> (which typically contains the class and detail message of <code>cause</code>).</p> <dl> <dt>Parameters:</dt> <dd>
<code>cause</code> - the cause (which is saved for later retrieval by the <a href="../../lang/throwable#getCause()"><code>Throwable.getCause()</code></a> method). (A <code>null</code> value is permitted, and indicates that the cause is nonexistent or unknown.)</dd> <dt>Since:</dt> <dd>1.5</dd> </dl>   </section>  </div>  </main>  <div class="_attribution">
  <p class="_attribution-p">
    &copy; 1993, 2020, Oracle and/or its affiliates. All rights reserved.<br>Documentation extracted from Debian's OpenJDK Development Kit package.<br>Licensed under the GNU General Public License, version 2, with the Classpath Exception.<br>Various third party code in OpenJDK is licensed under different licenses (see Debian package).<br>Java and OpenJDK are trademarks or registered trademarks of Oracle and/or its affiliates.<br>
    <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/security/cert/CertificateException.html" class="_attribution-link">https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/security/cert/CertificateException.html</a>
  </p>
</div>
